contracts/clients/src/OperationResults.ts

```diff
@@ -107,19 +107,20 @@ const getError = (
   return decodeError(errorData);
 };
 
+/**
+ * Gets the results of operations (and actions) run through VerificationGateway.processBundle.
+ * Decodes unsuccessful operations into an error message and the index of the action that failed.
+ *
+ * @param transactionReceipt Transaction receipt from a VerificationGateway.processBundle transaction
+ * @returns An array of decoded operation results
+ */
 export const getOperationResults = (
   txnReceipt: ContractReceipt,
 ): OperationResult[] => {
-  if (!txnReceipt.events || !txnReceipt.events.length) {
-    throw new Error(
-      `no events found in transaction ${txnReceipt.transactionHash}`,
-    );
-  }
-
-  const walletOpProcessedEvents = txnReceipt.events.filter(
+  const walletOpProcessedEvents = txnReceipt.events?.filter(
     (e) => e.event === "WalletOperationProcessed",
   );
-  if (!walletOpProcessedEvents.length) {
+  if (!walletOpProcessedEvents || !walletOpProcessedEvents.length) {
```

Comment:
Possibly a bit less readable but nice and concise
```suggestion
  if (!walletOpProcessedEvents?.length) {
```

Also maybe one for retro, but wondered what your thoughts on negative vs positive logic were? I don't have a strong opinion but think it can help readability sometimes - https://www.obsidiansecurity.com/blog/why-to-stop-writing-negative-code/

Reply:
I think I somewhat agree with the difficulty of parsing negative logic. However, it is really helpful in reducing nesting in branching statements via early returns. For example:

```typescript
if (isCar) {
    if (isSedan) {
        return "Sedan";
    } else {
        return "Car";
    }
} else {
    return "Unknown";
}
```

vs.

```typescript
if (!isCar) {
    return "Unknown";
}
if (!isSedan) {
    return "Car";
}
return "Sedan";
```

This is a contrived example, but you can imagine how more complicated branching or error checking/validation could get difficult to grok.

Reply:
Yeah makes sense regarding nesting. I don't have a strong opinion on it but just wanted to get your thoughts :)